extensions/amp-story/1.0/amp-story.js

```diff
@@ -445,12 +446,45 @@ export class AmpStory extends AMP.BaseElement {
 
   /** @private */
   initializeStyles_() {
+    const mediaQueryEls =
+        this.element.querySelectorAll('media-query');
+
+    if (mediaQueryEls.length) {
+      this.initializeMediaQueries_(mediaQueryEls);
+    }
+
     const styleEl = document.querySelector('style[amp-custom]');
-    if (!styleEl) {
-      return;
+
+    if (styleEl) {
+      this.rewriteStyles_(styleEl);
     }
+  }
 
-    this.rewriteStyles_(styleEl);
+  /**
+   * Registers the media queries
+   * @param {!NodeList<!Element>} mediaQueryEls
+   * @private
+   */
+  initializeMediaQueries_(mediaQueryEls) {
+    const service = getMediaQueryService(this.getAmpDoc());
+
+    const onMediaQueryMatch = (matches, attribute) => {
+      this.mutateElement(() => {
+        matches ?
+          this.element.setAttribute(attribute, '') :
```

Comment:
Sorry, I thought this was going to be a class name?  I think it's more semantic and also we'll have less likelihood to break people in the future by adding new attributes